docs/modules/ROOT/pages/erc1155-pausable.adoc

```diff
@@ -0,0 +1,129 @@
+
+[[usage]]
+== Usage
+
+In order to make your ERC1155 token `pausable`, you need to use the https://docs.rs/openzeppelin-stylus/0.2.0-alpha.2/openzeppelin_stylus/utils/pausable/index.html[`Pausable`] contract and apply its mechanisms to ERC1155 token functions as follows:
+
+[source,rust]
+----
+use openzeppelin_stylus::{
+    token::erc1155::{Erc1155, IErc1155},
+    utils::Pausable,
+};
+
+sol_storage! {
+    #[entrypoint]
+    struct Erc1155Example {
+        #[borrow]
+        Erc1155 erc1155;
+        #[borrow]
+        Pausable pausable;
+    }
+}
+
+#[public]
+#[inherit(Erc1155, Pausable)]
+impl Erc1155Example {
+    pub fn mint(
+        &mut self,
+        to: Address,
+        token_id: U256,
+        amount: U256,
+        data: Bytes,
+    ) -> Result<(), Vec<u8>> {
+        // ...
+        self.pausable.when_not_paused()?;
+        // ...
+        self.erc1155._mint(to, token_id, amount, &data)?;
+        Ok(())
+    }
+
+    pub fn mint_batch(
+        &mut self,
+        to: Address,
+        token_ids: Vec<U256>,
+        amounts: Vec<U256>,
+        data: Bytes,
+    ) -> Result<(), Vec<u8>> {
+        // ...
+        self.pausable.when_not_paused()?;
+        // ...
+        self.erc1155._mint_batch(to, token_ids, amounts, &data)?;
+        Ok(())
+    }
+
+    pub fn burn(
+        &mut self,
+        account: Address,
+        token_id: U256,
+        value: U256,
+    ) -> Result<(), Vec<u8>> {
+        // ...
+        self.pausable.when_not_paused()?;
+        // ...
+        self.erc1155.burn(account, token_id, value)?;
+        Ok(())
+    }
+
+    pub fn burn_batch(
+        &mut self,
+        account: Address,
+        token_ids: Vec<U256>,
+        values: Vec<U256>,
+    ) -> Result<(), Vec<u8>> {
+        // ...
+        self.pausable.when_not_paused()?;
+        // ...
+        self.erc1155.burn_batch(account, token_ids, values)?;
+        Ok(())
+    }
+
+    pub fn safe_transfer_from(
+        &mut self,
+        from: Address,
+        to: Address,
+        id: U256,
+        value: U256,
+        data: Bytes,
+    ) -> Result<(), Vec<u8>> {
+        // ...
+        self.pausable.when_not_paused()?;
+        // ...
+        self.erc1155.safe_transfer_from(from, to, id, value, data)?;
+        Ok(())
+    }
+
+    pub fn safe_batch_transfer_from(
+        &mut self,
+        from: Address,
+        to: Address,
+        ids: Vec<U256>,
+        values: Vec<U256>,
+        data: Bytes,
+    ) -> Result<(), Vec<u8>> {
+        // ...
+        self.pausable.when_not_paused()?;
+        // ...
+        self.erc1155.safe_batch_transfer_from(from, to, ids, values, data)?;
+        Ok(())
+    }
+}
+----
+
+Additionally, you need to ensure proper initialization during xref:deploy.adoc[contract deployment]. Make sure to include the following code in your Solidity Constructor:
+
+[source,solidity]
+----
+contract Erc1155Example {
+    bool private _paused;
+
+    constructor() {
+        _paused = false;
```

Comment:
I don't think you need to set this field explicitly in the constructor, as it will be `false` by default.

I ran tests without it in [constructor.sol](https://github.com/OpenZeppelin/rust-contracts-stylus/pull/432/files#diff-7f3c841642ef646323e71ab520e66e456d325ddc367e3ba17b2fb6a118b81451R13), and they ran successfully

Reply:
We have `_paused = false` in other constructors so let's be consistent.